src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -149,17 +161,27 @@ private async Task ProcessOutgoingMessagesAsync(ServiceConnectionContext connect
             }
         }
 
-        private void AddClientConnection(ServiceConnectionContext connection, string instanceId)
+        private void AddClientConnection(ServiceConnectionContext connection, OpenConnectionMessage message)
         {
+            var instanceId = GetInstanceId(message.Headers);
+
             _clientConnectionManager.AddClientConnection(connection);
             _connectionIds.TryAdd(connection.ConnectionId, instanceId);
         }
 
         protected override Task OnConnectedAsync(OpenConnectionMessage message)
         {
             var connection = _clientConnectionFactory.CreateConnection(message, ConfigureContext);
-            AddClientConnection(connection, GetInstanceId(message.Headers));
-            Log.ConnectedStarting(Logger, connection.ConnectionId);
+            AddClientConnection(connection, message);
```

Comment:
why change to pass in `message`?

Reply:
Because we have to read `message.Headers` while creating the connection.
Our migration flag defined inside of it.

Reply:
I don't see related changes?

---
In reply to: [347766851](https://github.com/Azure/azure-signalr/pull/739#discussion_r347766851) [](ancestors = 347766851)